#! /bin/sh -e

# DP: libgcj_bc

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0

2008-04-16  Andrew Haley  <aph@littlepinkcloud.com>

	* libgcj_bc.c: (_Jv_LookupInterfaceMethod, _Jv_MonitorExit,
	_Jv_RegisterResource, _Jv_JNI_PopSystemFrame): Add.

--- libjava/libgcj_bc.c~	2008-04-18 09:50:42.194121373 +0200
+++ libjava/libgcj_bc.c	2008-04-18 09:53:17.936470710 +0200
@@ -33,7 +33,9 @@
 void _Jv_ResolvePoolEntry () {}
 void _Jv_Throw () {}
 void _Jv_MonitorEnter () {}
+void _Jv_MonitorExit () {}
 void _Jv_NewPrimArray () {}
+void _Jv_RegisterResource () {}
 void _Jv_NewObjectArray () {}
 void _Jv_NewMultiArray () {}
 void _Jv_ThrowBadArrayIndex () {}
@@ -43,6 +45,7 @@
 void _Jv_CheckCast () {}
 void _Jv_IsInstanceOf () {}
 void _Jv_CheckArrayStore () {}
+void _Jv_LookupInterfaceMethod () {}
 void _Jv_LookupInterfaceMethodIdx () {}
 
 void _Jv_RegisterClasses () 
@@ -66,6 +69,7 @@
 void _Jv_LookupJNIMethod () {}
 void _Jv_GetJNIEnvNewFrame () {}
 void _Jv_UnwrapJNIweakReference () {}
+void _Jv_JNI_PopSystemFrame () {}
 
 
 /* Checked divide (-fuse-divide-subroutine).  */
